lib/matplotlib/inset.py

```diff
@@ -0,0 +1,258 @@
+                ex = 1 - ex
+            if self.axes.yaxis.get_inverted():
+                ey = 1 - ey
+            xy_data = x + ex * width, y + ey * height
+            if existing is None:
+                # Create new connection patch with styles inherited from the
+                # parent artist.
+                p = ConnectionPatch(
+                    xyA=xy_inset_ax, coordsA=self._inset_ax.transAxes,
+                    xyB=xy_data, coordsB=self.axes.transData,
+                    arrowstyle="-",
+                    edgecolor=self._edgecolor, alpha=self.get_alpha(),
+                    linestyle=self._linestyle, linewidth=self._linewidth)
+                self._connectors.append(p)
+            else:
+                # Only update positioning of existing connection patch.  We
+                # do not want to override any style settings made by the user.
+                existing.xy1 = xy_inset_ax
+                existing.xy2 = xy_data
+                existing.coords1 = self._inset_ax.transAxes
+                existing.coords2 = self.axes.transData
+
+        if existing is None:
+            # decide which two of the lines to keep visible....
+            pos = self._inset_ax.get_position()
+            bboxins = pos.transformed(self.get_figure(root=False).transSubfigure)
+            rectbbox = transforms.Bbox.from_bounds(x, y, width, height).transformed(
+                self._rectangle.get_transform())
+            x0 = rectbbox.x0 < bboxins.x0
+            x1 = rectbbox.x1 < bboxins.x1
+            y0 = rectbbox.y0 < bboxins.y0
+            y1 = rectbbox.y1 < bboxins.y1
+            self._connectors[0].set_visible(x0 ^ y0)
+            self._connectors[1].set_visible(x0 == y1)
+            self._connectors[2].set_visible(x1 == y0)
+            self._connectors[3].set_visible(x1 ^ y1)
+
+    @property
+    def rectangle(self):
+        """`.Rectangle`: the indicator frame."""
+        return self._rectangle
+
+    @property
+    def connectors(self):
+        """
+        4-tuple of `.patches.ConnectionPatch`
+            The four connector lines connecting to (lower_left, upper_left,
+            lower_right upper_right) corners of *inset_ax*. Two lines are
+            set with visibility to *False*,  but the user can set the
+            visibility to True if the automatic choice is not deemed correct.
+        """
+        if self._inset_ax is None:
+            return
+
+        if self._auto_update_bounds:
+            self._rectangle.set_bounds(self._bounds_from_inset_ax())
+        self._update_connectors()
+        return tuple(self._connectors)
+
+    def draw(self, renderer):
+        # docstring inherited
+        conn_same_style = []
+
+        # Figure out which connectors have the same style as the box, so should
+        # be drawn as a single path.
+        for conn in self.connectors or []:
+            if conn.get_visible():
+                drawn = False
+                for s in _shared_properties:
```

Comment:
Isn't it also the case that if the non-shared properties differ, then you can't merge the paths? For example, `zorder` isn't listed as shared, but might make a difference.

Reply:
Yeah, I wasn't super-happy about doing this with a hard-coded list, but so far haven't thought of a better way.  Very happy to take suggestions/pointers.

... oh I just realised I could use `ArtistInspector.get_setters` to make a complete list - maybe that's part of the solution, though we'd have to actively skip comparing some properties like _transform_ :thinking:

Reply:
I also just realised that I currently have the _zorder_ going to `Rectangle` in the `__init__` when it should be getting applied to `InsetIndicator`.  So I need a better way to keep the properties straight at that end too.